analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -71,121 +83,218 @@
     bicycle_parking.id IS NULL
 """
 
+
 sql30 = """
-CREATE TEMP TABLE oneway AS
+-- Collect all oneways
+CREATE TEMP TABLE oneway_highway AS
 SELECT
   id,
-  t.nid,
+  CASE -- If oneway=-1, pretent the way is the other way around
+    WHEN tags?'oneway' AND tags->'oneway' = '-1' THEN array_reverse(nodes)
+    ELSE nodes
+  END AS nodes,
+  linestring,
+  array_length(nodes, 1) AS length,
+  generate_subscripts(nodes, 1) AS nid_index
+FROM
+  highways
+WHERE
+  (is_oneway OR is_roundabout) AND
+  NOT is_area AND
+  NOT is_construction
+"""
+
+sql31 = """
+CREATE INDEX idx_oneway_highway_linestring ON oneway_highway USING GIST(linestring);
+"""
+
+sql32 = """
+-- Collect all nodes that could be possible (valid) end nodes
+CREATE TEMP TABLE allowed_end_nodes AS
+SELECT DISTINCT
+  unnest(nodes) AS id
+FROM
+  ways
+WHERE
+  ways.tags != '' AND
```

Comment:
Keep `::hstore`, doesn'it ?

Reply:
Yes, right, should've been `''::hstore`. Will include it in the next commit